### ISSUE TITLE
Mic muting icon when a player joins/leaves

### ISSUE THREAD

**Issue by shawndibble** [luvolondon/fvtt-module-jitsiwebrtc#17]
In some cases the muted mic icon will show when a user joins/leaves however my mic isn't really muted as push to talk still works fine.

When I click the mic button, the icon stays, however I am then actually muted. Then I have to click it again for the icon to go away and for my mic to unmute.

This does not happen to all users.

**Comment by dolence** [comment]
I can confirm this.

**Comment by bekriebel** [comment]
I've noticed this too, and I see in the code why it is happening. I'm hoping to resolve this along with some other adjustments to the code that I'm working on.

**Comment by luvolondon** [comment]
I´m having a hard time to reproduce this. Could you describe the scenario where this happens? Where does the icon show up in a false state, locally at the player´s browser or for the other players? Which player has "Push to talk" active?
What I´m seeing is: Firefox, player has "Push to talk" active, when reloading Mic icon is _always_ red. First click on mute mic has no change of mic icon, 2nd click then shows the white mic icon. This does not happen in Chrome.

**Comment by chet-manley** [comment]
I can reliably repro this in Chrome 84.0.4147.125 and FireFox 78.0.2 by literally just logging in as any non-GM user. 
- Red mic icon only shows up on the video UI element of the user who just logged in (only visible to that user)
- Enable/disable icon and tooltip are reversed (displays opposite of actual status)
- Clicking the enable/disable icon twice clears the red mic icon
- The user's mic is _not_ muted, despite the indicator
- None of the above will occur if you are the GM user
- Disabling the module resolves the issue

This also occurs when the GM blocks then unblocks a user's audio (that user will continue to see their red icon). Speaking of indicator icons in relation to this module...
- If the GM blocks a user's audio, the GM's interface does _not_ display the red mic for that user (making it really easy for the GM to forget they blocked someone's feed, and inconsistent with the behavior of blocking a user's video)
- When a user is blocked or muted, pressing the PTT button still changes their video border to solid (in their interface)
- If a user's audio is blocked by the GM, they can still toggle mute on and off in their interface (stock VTT seems to get around this by hiding the enable/disable icon)
- If a user's video is blocked by the GM, the only indicator (for the user) is their local-live feed disappearing, instead of seeing a red camera icon
- If a user disables their own video, their video feed UI element shrinks to the minimum size, regardless of the size set for the rest of the videos, and the red camera icon is missing

**Comment by luvolondon** [comment]
Hi,
thank you very much for this detailed list of steps to reproduce the problem. 
The correct handling of icon status has been addressed by a major rework done by @bekriebel , which I think is soon to be released. Let´s see if that also help here.

**Comment by bekriebel** [comment]
I believe this is fixed in the latest release (v0.3.0), can someone please confirm?

**Comment by chet-manley** [comment]
I'll try to test again sometime this weekend. We were having so many issues with video in VTT, we just disabled AV completely and moved to our FB group chat. Since most of our external members are playing on laptops, having the video in-window is still important to me, so I want to get this working.

**Comment by bekriebel** [comment]
I'm going to go ahead and close this issue. If anything issues still persist, please let me know!

**Comment by chet-manley** [comment]
We attempted to use this again at the start of our game last weekend. Video and audio was broken for 4 of our 7 players, so we ditched it again and went back to FB Messenger. Not sure if the merge fixed the icon issues, but we couldn't get anything to work.

**Comment by bekriebel** [comment]
There are several factors that may cause this such as player permission settings and the server you were attempting to use. If you'd like to continue trying to debug, feel free to open up a new issue with as many details as you have or reach out to me on the FVTT Discord - I'm `bekit` there.
